src/ipc/ipc4/handler.c

```diff
@@ -460,49 +460,55 @@ static int ipc4_load_library(struct ipc4_message_request *ipc4)
 
 static int ipc4_process_chain_dma(struct ipc4_message_request *ipc4)
 {
-	struct ipc4_chain_dma cdma;
+#if CONFIG_COMP_CHAIN_DMA
+	struct ipc_comp_dev *cdma_comp;
 	struct ipc *ipc = ipc_get();
-	bool delay = false;
-	int ret = memcpy_s(&cdma, sizeof(cdma), ipc4, sizeof(*ipc4));
+	struct ipc4_chain_dma cdma;
+	int comp_id;
+	int ret;
 
+	ret = memcpy_s(&cdma, sizeof(cdma), ipc4, sizeof(*ipc4));
 	if (ret < 0)
 		return IPC4_FAILURE;
 
-	if (cdma.primary.r.allocate && cdma.extension.r.fifo_size) {
-		ret = ipc4_create_chain_dma(ipc, &cdma);
-		if (ret) {
-			tr_err(&ipc_tr, "failed to create chain dma %d", ret);
-			return ret;
-		}
+	comp_id = IPC4_COMP_ID(cdma.primary.r.host_dma_id + IPC4_MAX_MODULE_COUNT, 0);
+	cdma_comp = ipc_get_comp_by_id(ipc, comp_id);
 
-		/* if enable is not set, chain dma pipeline is not going to be triggered */
-		if (!cdma.primary.r.enable)
-			return ret;
-	}
+	if (!cdma_comp && cdma.primary.r.allocate && cdma.primary.r.enable) {
+		ret = ipc4_create_mng_for_chain(&cdma);
```

Comment:
`ipc4_chain_manager_create()` seems to be a better name for this, but it's already there